src/astrobleme/logic/GameLoop.java

```diff
@@ -37,30 +37,39 @@
     private final GraphicsContext context;
     private float interval;
     private boolean running;
-    private boolean paused;
-    private boolean keyIsPressed;
+    private boolean stopped;
+    public boolean keyIsPressed;
 
     public GameLoop(final Grid grid, final GraphicsContext context) {
         this.grid = grid;
         this.context = context;
         interval = 1000.0f / FRAME_RATE; // 1000 ms in a second
         running = true;
-        paused = false;
+        stopped = false;
         keyIsPressed = false;
     }
 
     @Override
     public void run() {
-        while (running && !paused) {
+        while (true) {
             // Time the update and paint calls
+            if (running);
+            else {
+//                System.out.println("Pausing...... ");
+                try {
+                    Thread.sleep((long) (interval));
+                } catch (InterruptedException ignore){}
+                continue;
+            }
```

Comment:
Can you briefly explain what this does?

Reply:
It is for pause purpose, invoke by keypad 'P'. The event controlls the variable value running.